kuma/search/views.py

```diff
@@ -37,6 +40,17 @@ def search(request, *args, **kwargs):
 wiki_search = SearchView.as_view()
 
 
+class SearchRedirectView(RedirectView):
+    permanent = True
+
+    def get_redirect_url(self, *args, **kwargs):
+        get = self.request.GET.get('q', None)
```

Comment:
Instead of handling specific named keys from the query string, see if you can use something from `self.request.META`. I can't remember what it's called. Perhaps it's `self.request.META.get('QUERY_STRING')`. 
You can find out by doing a `print(self.request.META.keys())`

Reply:
Made the suggested change, works great. Thanks @peterbe!